### PR TITLE
Add calcIOEdgeBuildings

### PR DESCRIPTION
**Pull request by pfuehrlich-pik**
Adds calcIOEdgeBuildings which is doing the same as mrremind::calcIO for subtype output_EDGE or output_EDGE_buildings . In a separate upcoming mrremind PR these subtypes are removed from mrremind::calcIO. This calcIO split up allows mredgebuildings to only depend on mrcommons instead of mrremind.

### CODE REVIEW

**Review by rkrekeler — APPROVED** [review]

File: R/calcIOEdgeBuildings.R [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  # Add the values from bioshare to the other modern and traditional biomass
+  reminditems[, , nBiotrad] <- reminditems[, , nBiotrad] + bioshareTrad
+  reminditems[, , nBiomod] <- reminditems[, , nBiomod] + bioshareMod
+
+  # Remove the bioshare item
+  reminditems <- reminditems[, , nBioshare, invert = TRUE]
+
+  # TODO more (compared to mrremind calcIO) specific description?
```
Comment (rkrekeler) [inline R/calcIOEdgeBuildings.R:101]: suggested description
```
description = "Historic final energy demand from buildings (and industry) based on the 2017 IEA World Energy Balances"
```

File: R/calcIOEdgeBuildings.R [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    nBioshare <- "bioshare"
+  }
+  # Read-in data to compute income per capita
+  gdp <- calcOutput("GDPPast", aggregate = FALSE)
+  pop <- calcOutput("PopulationPast", aggregate = FALSE)
+  gdppop <- gdp[, intersect(getYears(gdp), getYears(pop)), ] / pop[, intersect(getYears(gdp), getYears(pop)), ]
+  # Create a lambda which is 1 for income per capita <= 10000, and 0 above 15000
+  # the multiplication by gdppop was necessary to avoid error from vector length.
+  lambda <- pmin(gdppop * 0 + 1, pmax(0 * gdppop, (15000 - gdppop) / (15000 - 10000)))
+  lambda <- time_interpolate(lambda, getYears(reminditems), extrapolation_type = "constant")
```
Comment (LaviniaBaumstark) [inline R/calcIOEdgeBuildings.R:75]: this part is really only EDGE related and speaks for the split

File: R/calcIOEdgeBuildings.R [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+           mapping <- toolGetMapping(type = "sectoral", name = "structuremappingIO_outputs.csv", returnPathOnly = TRUE)
+           target <- "EDGEitems"
+         },
+         output_EDGE_buildings = {
+           mapping <- toolGetMapping(type = "sectoral", name = "structuremappingIO_outputs.csv", returnPathOnly = TRUE)
+           target <- "EDGE_buildings"
+         })
+
+  # read in data and convert from ktoe to EJ
+  data <- readSource("IEA", subtype = "EnergyBalances") * 0.0000418680000
```
Comment (LaviniaBaumstark) [inline R/calcIOEdgeBuildings.R:32]: this speaks against splitting out the EDGE-part.

**Review by LaviniaBaumstark — COMMENTED** [review]
I would assume that the best solution would be to write a new calc-function which polishes the IEA data. Most of the subtype-specific calculations should word for all subtypes (they are only mandatory for the subtypes). In the end a caldIOEdgeBuildings still seems to be useful for some of the calculations.

**Pull request merged**